src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AMOUNT_CHARACTERS = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        checkCopy(user);
+        return storageDao.add(user);
+    }
+
+    private boolean checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RegisterException("User's login cant be null!");
+        } else if (user.getLogin().length() < MIN_AMOUNT_CHARACTERS) {
+            throw new RegisterException("User's login length = " + user.getLogin().length()
+                    + ", but min length = " + MIN_AMOUNT_CHARACTERS);
+        }
+        return true;
+    }
+
+    private boolean checkAge(User user) {
+        if (user.getAge() == null) {
+            throw new RegisterException("User's age cant be null!");
+        } else if (user.getAge() < MIN_AGE) {
```

Comment:
you don't need to wrap the logic that goes after throwing exception in `else` block. We won't reach in any other way